### PR TITLE
[GPU] add new unsqueeze mode

### PR DESCRIPTION
**Pull request by andrei-cv**
### Details:
 - *add new axis mode*

### Tickets:
 - *CVS-105897, CVS-129991*

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl [hunk 1]
```diff
@@ -289,33 +289,68 @@ KERNEL(strided_slice_ref)(OPTIONAL_SHAPE_INFO_ARG
...
+    const uint x = yx % OUTPUT_SIZE_X;
+#elif OUTPUT_LAYOUT_BFWZYX
+    const uint zyx = (uint)get_global_id(2) % (OUTPUT_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_Z);
+    const uint w = (uint)get_global_id(2) / (OUTPUT_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_Z);
+    const uint z = zyx / (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
+    const uint yx = zyx % (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
+    const uint y = yx / OUTPUT_SIZE_X;
+    const uint x = yx % OUTPUT_SIZE_X;
+#endif
+    const uint output_index = OUTPUT_OFFSET +
```
Comment (Lyamin-Roman) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl:345]: Use OUTPUT_GET_INDEX instead

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl [hunk 2]
```diff
@@ -289,33 +289,68 @@ KERNEL(strided_slice_ref)(OPTIONAL_SHAPE_INFO_ARG
...
+    
     const uint input_index = INPUT0_OFFSET +
         batch * INPUT0_BATCH_PITCH +
-        feature * INPUT0_FEATURE_PITCH +
-        w_input * INPUT0_W_PITCH +
-        z_input * INPUT0_Z_PITCH +
-        y_input * INPUT0_Y_PITCH +
-        x_input * INPUT0_X_PITCH;
-    output[input_index] = input[input_index];
+        input_feature_id * INPUT0_FEATURE_PITCH +
```
Comment (Lyamin-Roman) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl:320]: Use INPUT0_GET_INDEX(batch, input_feature_id ...)  instead

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl [hunk 3]
```diff
@@ -289,33 +289,68 @@ KERNEL(strided_slice_ref)(OPTIONAL_SHAPE_INFO_ARG
...
     const uint w_input = 0;
     const uint z_input = 0;
-    const uint y_input = (uint)get_global_id(2) / INPUT0_SIZE_X;
-    const uint x_input = (uint)get_global_id(2) % INPUT0_SIZE_X;
-#elif OUTPUT_LAYOUT_BFZYX
+    const uint y_input = index_in_batch / OUTPUT_SIZE_X;
+    const uint x_input = index_in_batch % OUTPUT_SIZE_X;
+#elif INPUT0_LAYOUT_BFZYX
+    const uint index_in_batch = (feature * get_global_size(2) +(uint)get_global_id(2))%(OUTPUT_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_Z);
+    uint input_feature_id = (feature * get_global_size(2) +(uint)get_global_id(2)) / (OUTPUT_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_Z);
```
Comment (Lyamin-Roman) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl:301]: nit: Fix code style according to the surrounding code
uint -> const uint
Spaces before and after operation signs

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl (hunk 2 above)
Comment (andrei-cv) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl:320]: It not equal INPUT0_GET_INDEX, use It use output and input0 pitches

**Review by Lyamin-Roman — APPROVED** [review]

**Pull request merged**